### ISSUE TITLE
[indent] Indent error not apply to `type`

### ISSUE THREAD

**Issue by rhalaly** [typescript-eslint/typescript-eslint#3707]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/indent": ["error", 2]
  }
}
```

```TS
const foo = 10;
                        type Bar = number;
const baz = 10;
```

**Expected Result**

Error on line 2 that the `type` definition is not indented correctly.

**Actual Result**

No errors at all

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.29.0` |
| `@typescript-eslint/parser`        | `4.29.0` |
| `TypeScript`                       | `4.2.3` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `15.12.0` |

**Comment by bradzacher** [comment]
Please use the search. There is a ****pinned**** issue which covers this.

**Comment by rhalaly** [comment]
Ho sorry, I missed the pined issue. I did search for a solution before...
